### PR TITLE
feat(amazon q transform): Human In The Loop

### PR DESCRIPTION
**Pull request by damntrecky**
## Problem

Users builds fail sometimes do to 1p dependencies and those resources are not available at the transformation level in the backend. We are adding an "human in the loop" intervention to allow users to choose a version and resume a transformation

## Solution

Display chat input fields, from maven output, to allow users to choose a 1p dependency upgrade version and continue on.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/commands/startTransformByQ.ts [hunk 1]
```diff
@@ -145,6 +157,15 @@ export async function startTransformByQ() {
     // Set the default state variables for our store and the UI
     await setTransformationToRunningState()
 
+    await completeHumanInTheLoopWork('fake-job-id', 0)
+    await postTransformationJob()
+    await cleanupTransformationJob(intervalId)
+
+    const quickExit = true
+    if (quickExit) {
```
Comment (mk-fan) [inline packages/core/src/codewhisperer/commands/startTransformByQ.ts:165]: What do these line do?

File: packages/core/src/codewhisperer/commands/startTransformByQ.ts [hunk 2]
```diff
@@ -268,9 +289,117 @@ export async function pollTransformationStatusUntilComplete(jobId: string) {
...
+        // 4) We need to run maven commands on that pom.xml to get available versions
+        const compileFolderInfo: FolderInfo = {
+            name: tmpDependencyListFolderName,
+            path: tmpDependencyListDir,
+        }
+        runMavenDependencyUpdateCommands(compileFolderInfo)
+        const { latestVersion, majorVersions, minorVersions } = await parseXmlDependenciesReport(
+            path.join(tmpDependencyListDir, localPathToXmlDependencyList)
+        )
+        console.log(latestVersion, majorVersions, minorVersions)
```
Comment (mk-fan) [inline packages/core/src/codewhisperer/commands/startTransformByQ.ts:359]: Remove the console.log?

File: packages/core/src/codewhisperer/commands/startTransformByQ.ts [hunk 3]
```diff
@@ -268,9 +289,117 @@ export async function pollTransformationStatusUntilComplete(jobId: string) {
...
+        const uploadFolderInfo: FolderInfo = {
+            name: userDependencyUpdateFolderName,
+            path: userDependencyUpdateDir,
+        }
+        runMavenDependencyBuildCommands(uploadFolderInfo)
+        // TODO modify code to be re-usable with current framework here
+        // TODO Update manifest.json file for upload
+        const uploadPayloadFilePath = await zipCode(uploadFolderInfo)
+        const uploadId = await uploadPayload(uploadPayloadFilePath)
+        console.log('Finished human in the loop work', uploadId)
```
Comment (mk-fan) [inline packages/core/src/codewhisperer/commands/startTransformByQ.ts:387]: Remove the console.log?

File: packages/core/src/amazonqGumby/mock/downloadHilZip/manifest.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (justinmk3) [inline packages/core/src/amazonqGumby/mock/downloadHilZip/manifest.json:1]: should this be in `packages/core/src/testFixtures/` , or somewhere else in  `src/test/`

File: packages/core/src/codewhisperer/service/amazonQGumby/transformByQApiHandler.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    const manifestFileVirtualFileReference = vscode.Uri.file(
+        '/Users/nardeck/workplace/gumby-prod/aws-toolkit-vscode/packages/core/src/amazonqGumby/mock/downloadHilZip/manifest.json'
+    )
+    const pomFileVirtualFileReference = vscode.Uri.file(
+        '/Users/nardeck/workplace/gumby-prod/aws-toolkit-vscode/packages/core/src/amazonqGumby/mock/downloadHilZip/pom.xml'
+    )
+    return { manifestFileVirtualFileReference, pomFileVirtualFileReference }
+}
+
+export async function getTransformationStepsFixture(
```
Comment (justinmk3) [inline packages/core/src/codewhisperer/service/amazonQGumby/transformByQApiHandler.ts:23]: this is in src/, should it be in src/test/

**Comment by justinmk3** [comment]
revert the package.json changes

**Review by justinmk3 — CHANGES_REQUESTED** [review]
revert the `packages/toolkit/package.json` changes

File: packages/core/src/amazonqGumby/telemetry/codeTransformTelemetryState.ts [hunk 6]
```diff
@@ -34,6 +43,8 @@ class CodeTransformerTelemetryState {
     public getSessionId = () => this.mainState.sessionId
     public getStartTime = () => this.mainState.sessionStartTime
     public getResultStatus = () => this.mainState.resultStatus
+    public getCodeTransformMetaData = () => this.mainState.codeTransformMetadata
+    public getCodeTransformMetaDataString = () => JSON.stringify(this.mainState.codeTransformMetadata)
```
Comment (justinmk3) [inline packages/core/src/amazonqGumby/telemetry/codeTransformTelemetryState.ts:47]: how big is this payload, potentially? and does it potentially contain PII?

**Review by justinmk3 — APPROVED** [review]

File: packages/core/src/amazonqGumby/telemetry/codeTransformTelemetryState.ts (hunk 6 above)
Comment (damntrecky) [reply packages/core/src/amazonqGumby/telemetry/codeTransformTelemetryState.ts:47]: The `codeTransformMetadata` is the start of the new metrics to switch away from the current telemetry implementation. We, we will add a safeguard for truncating this for the max column length in the backend. There is a whole internal proposal from our BI team that has been passed around a couple months back. We are testing it out for this project.

There is not any PII data, we are not logging any user. Only success and generic failure detail messages.

File: packages/core/src/amazonqGumby/chat/controller/messenger/messenger.ts [hunk 7]
```diff
@@ -328,6 +337,28 @@ export class Messenger {
...
+            new ChatMessage(
+                {
+                    message,
+                    messageType: 'ai-prompt',
+                },
+                tabID
+            )
+        )
+
+        this.sendInProgressMessage(tabID, 'Stopping job...')
```
Comment (JuliaABurch) [inline packages/core/src/amazonqGumby/chat/controller/messenger/messenger.ts:359]: This isn't needed anymore, but I'll remove this in a separate PR / when we're able to test UX again

File: packages/core/src/codewhisperer/client/codewhisperer.ts [hunk 8]
```diff
@@ -38,8 +38,8 @@ export interface CodeWhispererConfig {
 }
 
 export const defaultServiceConfig: CodeWhispererConfig = {
-    region: 'us-east-1',
-    endpoint: 'https://codewhisperer.us-east-1.amazonaws.com/',
+    region: 'us-west-2',
```
Comment (JuliaABurch) [inline packages/core/src/codewhisperer/client/codewhisperer.ts:41]: Revert this

Comment (damntrecky) [reply packages/core/src/codewhisperer/client/codewhisperer.ts:41]: DONE.

**Review by JuliaABurch — APPROVED** [review]

**Pull request merged**